src/rag/vector_store/pgvector.py

```
@@ -1,28 +1,322 @@
+                            f"""
+                            INSERT INTO {self.table_name}
+                                (doc_id, title, content, embedding, metadata)
+                            VALUES (%s, %s, %s, %s, %s)
+                            ON CONFLICT (doc_id)
+                            DO UPDATE SET
+                                title = EXCLUDED.title,
+                                content = EXCLUDED.content,
+                                embedding = EXCLUDED.embedding,
+                                metadata = EXCLUDED.metadata,
+                                updated_at = CURRENT_TIMESTAMP
+                            """,
```

Ruff (0.15.0)

[error] Possible SQL injection vector through string-based query construction

(S608)